gamecenter/api/views.py

```diff
@@ -15,6 +18,38 @@
 score_schema = ScoreSchema()
 
 
+def handle_api_key(f):
+    @functools.wraps(f)
+    def decorated_function(*args, **kwargs):
+        if "Authorization" in request.headers:
+            api_key = request.headers["Authorization"][7:]  # from the string form "Bearer <api_key>"
+        else:
+            api_key = ""
+        game = Game.query.filter(Game.api_key == api_key).first()
+        if game:
+            g.game = game
+            return f(*args, **kwargs)
+        else:
+            # todo raise AuthenticationError("Invalid api key")
+            return "Unable to authenticate you.", 401
```

Comment:
take a look here: https://github.com/Rdbaker/GameCenter/blob/master/gamecenter/core/utils.py#L5

if you raise that with a message and an error code, it will automatically return a json response with the error